vortex-array/src/compute/is_constant.rs

```diff
@@ -96,15 +106,19 @@ pub fn is_constant(array: &dyn Array) -> VortexResult<bool> {
             array.encoding()
         );
 
-        let array = array.to_canonical()?;
-
-        if let Some(is_constant_fn) = array.as_ref().vtable().is_constant_fn() {
-            is_constant_fn.is_constant(array.as_ref())?
+        if opts.fallback_to_canonicalize {
+            let array = array.to_canonical()?;
+
+            if let Some(is_constant_fn) = array.as_ref().vtable().is_constant_fn() {
+                is_constant_fn.is_constant(array.as_ref())?
+            } else {
+                vortex_bail!(
+                    "No is_constant function for canonical array: {}",
+                    array.as_ref().encoding(),
+                )
+            }
         } else {
-            vortex_bail!(
-                "No is_constant function for canonical array: {}",
-                array.as_ref().encoding(),
-            )
+            None
```

Comment:
We could log::debug a missing is_constant impl as with the other compute functions